blue_bringup/scripts/gripper_calibration_service.py

```diff
@@ -0,0 +1,143 @@
+            rospy.wait_for_message("blue_hardware/motor_states", MotorState)
+
+        # set simple calibration angles again to calibrate gripper
+        response = joint_startup_calibration(startup_positions, disable_snap)
+        if response.success:
+            rospy.loginfo("Joint startup calibration succeeded!")
+        else:
+            rospy.logerr("Joint startup calibration failed!")
+
+        cmd.data = [0.0]
+        # apply positive torque
+        for i in range(5):
+            cmd_pub.publish(cmd)
+            rate.sleep()
+
+        # switch out controllers to reset position
+        gripper_controller_response = switch_controller(['blue_controllers/gripper_controller'], ['blue_controllers/gripper_torque_controller'], 2)
+
+        client = actionlib.SimpleActionClient(
+            "blue_controllers/gripper_controller/gripper_cmd",
+            GripperCommandAction,
+        )
+
+        # Wait 10 Seconds for the gripper action server to start or exit
+        if not client.wait_for_server(rospy.Duration(10.0)):
+            rospy.logerr("Exiting - Gripper Action Server Not Found")
+            rospy.signal_shutdown("Action Server not found")
+            sys.exit(1)
+
+        goal = GripperCommandGoal()
+        goal.command.position = 0.0
+        goal.command.max_effort = 5.0
+        client.send_goal(goal)
+        client.wait_for_result()
+
+        # clear gripper controllers
+        gripper_controller_response = switch_controller([], ['blue_controllers/gripper_controller'], 2)
+        rospy.loginfo("Gripper Calibration Succeeded!")
+
+    except rospy.ServiceException as e:
+        rospy.logerr("Joint startup calibration failed: %s" % e)
+        return TriggerResponse(False, "Fail")
+
+    return TriggerResponse(True, "Succsess")
+
+if __name__ == "__main__":
+    global gripper_state
+    global motor_names
+    global disable_snap
+    global startup_positions
+
+    global joint_startup_calibration
+    global switch_controller
+    global cmd_pub
+
+    rospy.init_node("simple_startup_calibration")
+    rate = rospy.Rate(5)
+
+    # Read startup angles from parameter server
+    rospy.loginfo("Reading desired joint angles...")
+    startup_positions = rospy.get_param("blue_hardware/simple_startup_angles")
+    startup_positions[-1] = 1.05
+    motor_names = rospy.get_param("blue_hardware/motor_names")
+    disable_snap = rospy.get_param("blue_hardware/disable_snap", False)
+
+    # Wait for calibration service to come up
+    rospy.loginfo("Waiting for calibration service...")
+    rospy.wait_for_service('blue_hardware/joint_startup_calibration')
+    rospy.wait_for_service('controller_manager/switch_controller')
+
+    # Load Relavent Controllers
+    load_controller = rospy.ServiceProxy('controller_manager/load_controller', LoadController)
+    load_controller('blue_controllers/gripper_torque_controller')
+    load_controller('blue_controllers/gripper_controller')
```

Comment:
would add complexity, but i think this would be cleaner if you (a) handled controller loading when the service is called and (b) unloaded the controllers when you're done

some other things to think about:
- what happens if the controllers are already loaded? don't unload at the end?
- what happens if the needed controllers have already been loaded and started? return an error?

Reply:
Yes, that would be nice to handel